qt/app.py

```diff
@@ -69,267 +105,61 @@ def __init__(self, config, appInstance, qapp):
         self.appInstance = appInstance
         self.qapp = qapp
         self.snapshots = snapshots.Snapshots(config)
+
         self.lastTakeSnapshotMessage = None
         self.tmpDirs = []
+        self.firstUpdateAll = True
+        self.disableProfileChanged = False
 
-        #main toolbar
-        self.mainToolbar = self.addToolBar('main')
-        self.mainToolbar.setFloatable(False)
+        # "Magic" object handling shutdown procedure in different desktop
+        # environments.
+        self.shutdown = tools.ShutDown()
 
-        #window icon
+        # Import on module level not possible because of Qt restrictions.
         import icon
+        globals()['icon'] = icon
```

Comment:
That is a workaround until we implement a solution to our icon handling problem. I'll wait on aryodas trayicon-fix until that.

Reply:
THX for waiting, my change is merged, your way is clear now :-)